stacktrace: add version 1.0.0

AiyionPrime opened the pull request:
### Summary
Changes to recipe: **stacktrace/[1.0.0]**

#### Motivation
I'd like to get rid of a submodule in our project.
For most of my dependencies there existed conan2 packages, this one is lacking.

The header is small, not expected to change anymore and not present in git or zip-form.

#### Details

In 2008 Timo Bingmann wrote a useful helper to demangle gcc/clang compilations and did a writeup: https://panthema.net/2008/0901-stacktrace-demangled/

I really appreciate the size/usefulness ratio and would like to see it in the CCI.

There is currently no CCI issue this PR fixes, as I did read the guide after implementing the recipe. I can open an issue if desired, just feel like we can discuss the contribution here as well.

_I'm a first time contributor, and though I read the guidelines and the packaging tutorial increased attention to detail while reviewing might be in order :)_

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan **2.12.2**

jcar87 commented:
Hi @AiyionPrime - apologies we didn't get to you in time. We review PRs before running on our CI for first time contributors (or past contributors after long inactivity) - when we don't, we should leave a message about this, apologies that this didn't happen in time.

It is a bit unusual for us to add libraries where the code doesn't come from a canonical location that the recipe references for download. Including external code can tangle the licensing of the repository (we are not the authors), and confuse the maintenance story: we don't maintain the libraries, we maintain the recipes. We typically look at this being hosted and maintained - and look for evidence of usage elsewhere (e.g. stars on github, availability on other linux distros, etc).  Do you know if the latter is the case, to make a compelling case? Either way - we would require the sources to be available externally in a canonical location.

AiyionPrime commented:
@jcar87 Alright, I found the `download` tool which appears to be the right choice for the job.
I do understand a git repo would be nicer, but given the lack of changes in the past and the fixed sha256sum, I'd say this suffices for now.

I cleaned up the conanfile, rebased this onto master and hope this is now fit for a first inspection. Have a nice weekend everyone!

AiyionPrime commented:
@jcar87 I addressed the two issues the linter found, but I think this needs to be validated again?

AiyionPrime commented:
@jcar87 Thanks :)
Anything is there anything left to do here for me?
